benchmarks/persistence/persistence.py

```diff
@@ -0,0 +1,229 @@
+        workflow run identifier.
+
+        Args:
+            object: The object to be persisted.
+        """
+        client = self._object_session.client("s3")
+
+        client.put_object(
+            Bucket=self._bucket_name,
+            Key=object.get_full_file_path(),
+            Body=json.dumps(object.get_object()),
+            ContentType="application/json",
+        )
+
+
+@define
+class LocalFileSystemObjectCreator:
+    """Class for creating the path of a result object."""
+
+    _benchmark: Benchmark = field(validator=instance_of(Benchmark))
+    """The benchmark for which the result is stored."""
+
+    _result: Result = field(validator=instance_of(Result))
+    """The result of the benchmark."""
+
+    _path: str = field(validator=instance_of(str))
+    """The path where the result is stored."""
+
+    _filename: str = field(validator=instance_of(str))
+    """The filename of the result object."""
+
+    @_path.default
+    def _default_bucket_name(self) -> str:
+        ENVIRONMENT_NOT_SET = "BAYBE_BENCHMARK_PERSISTENCE_PATH" not in os.environ
+        if ENVIRONMENT_NOT_SET:
+            return "./"
+        return os.environ["BAYBE_BENCHMARK_PERSISTENCE_PATH"]
+
+    @_filename.default
+    def _default_filename(self) -> str:
+        experiment_identifier = self._benchmark.name
+        metadata = self._result.metadata
+        file_usable_date = metadata.start_datetime.strftime("%Y-%m-%d")
+        return (
+            f"{experiment_identifier}_{metadata.latest_baybe_tag}"
+            + f"_{file_usable_date}_{metadata.commit_hash}.json"
+        )
+
+    def get_full_file_path(self) -> str:
+        """Construct the path of a result object.
+
+        Returns:
+            The path of the result object.
+        """
+        return f"{self._path}/{self._filename}"
+
+    def get_object(self) -> dict:
+        """Construct the object to be persisted.
+
+        Returns:
+            The object to be persisted.
+        """
+        return self._result.to_dict() | self._benchmark.to_dict()
+
+
+@define
+class LocalFileSystemPersistenceHandler:
+    """Class for persisting experiment results in an S3 bucket."""
+
+    def write_object(self, object: PersistenceObjectInterface) -> None:
+        """Store the result of a benchmark.
+
+        This method will store the result of a benchmark in an S3 bucket.
+        The key of the object will be the experiment identifier, the branch, the
+        BayBE-version, the start datetime, the commit hash and the workflow
+        run identifier.
+
+        Args:
+            object: The object to be persisted.
+        """
+        with open(object.get_full_file_path(), "w") as file:
+            json.dump(object.get_object(), file)
+
+
+def persister_factory() -> PersistenceHandlingInterface:
+    """Create a persistence handler based on the environment variables.
+
+    Returns:
+        The persistence handler.
+    """
+    if "GITHUB_RUN_ID" in os.environ:
```

Comment:
We should somehow store the `GITHUB_RUN_ID` as a variable to avoid having all these hard-coded string literals lying around

Reply:
Agree with you. Normally I would avoid to have a global constant but since it is some kind of a state description considering the execution environment, it might be okay.